apps/api/src/variable/service/variable.service.ts

```diff
@@ -413,49 +413,79 @@ export class VariableService {
     sort: string,
     order: string,
     search: string
-  ) {
-    // Check if the user has the required authorities in the project
-    await this.authorityCheckerService.checkAuthorityOverProject({
-      userId: user.id,
-      entity: { id: projectId },
-      authority: Authority.READ_VARIABLE,
-      prisma: this.prisma
-    })
+  ): Promise<
+    { environmentName: string; environmentId: string; variables: any[] }[]
+  > {
+    try {
+      // Check if the user has the required authorities in the project
+      await this.authorityCheckerService.checkAuthorityOverProject({
+        userId: user.id,
+        entity: { id: projectId },
+        authority: Authority.READ_VARIABLE,
+        prisma: this.prisma
+      })
 
-    return await this.prisma.variable.findMany({
-      where: {
-        projectId,
-        pendingCreation: false,
-        name: {
-          contains: search
-        }
-      },
-      include: {
-        versions: {
-          orderBy: {
-            version: 'desc'
-          },
-          take: 1
+      const variables = await this.prisma.variable.findMany({
+        where: {
+          projectId,
+          pendingCreation: false,
+          name: {
+            contains: search
+          }
         },
-        lastUpdatedBy: {
-          select: {
-            id: true,
-            name: true
+        include: {
+          versions: {
+            orderBy: {
+              version: 'desc'
+            },
+            take: 1
+          },
+          lastUpdatedBy: {
+            select: {
+              id: true,
+              name: true
+            }
+          },
+          environment: {
+            select: {
+              id: true,
+              name: true
+            }
           }
         },
-        environment: {
-          select: {
-            id: true,
-            name: true
+        skip: page * limit,
+        take: limit,
+        orderBy: {
+          [sort]: order
+        }
+      })
+
+      // Group variables by environment
+      const variablesByEnvironment: {
+        [key: string]: {
+          environmentName: string
```

Comment:
This works fine! Just change `environment` field like I mentioned in secret service.